mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py

```diff
@@ -73,28 +83,58 @@ def _state_name(self):
         readonly=True,
         default="NEW"
     )
-    date = fields.Date(
-        'Date',
+
+    state_id = fields.Many2one(
+        'mgmtsystem.nonconformity.state',
+        'State',
+        default=_default_state)
```

Comment:
States are used in write logic, so it is easier if you stick to them.
IMO you should keep states as a selection list, as you had before.